core/src/test/scala/unit/kafka/server/ReplicaManagerTest.scala

```diff
@@ -1323,6 +1323,17 @@ class ReplicaManagerTest {
 
       replicaManager.becomeLeaderOrFollower(2, leaderAndIsrRequest, (_, _) => ())
 
+      appendRecords(replicaManager, tp0, TestUtils.singletonRecords(s"message".getBytes)).onFire { response =>
+        assertEquals(Errors.NONE, response.error)
+      }
+      // fetch as follower to advance the high watermark
```

Comment:
nit: Should we say `Fetch as follower to initialise the log end offset of the replica`? Because this is what we need in the condition that you mentioned.